tensorboard/backend/application.py

```diff
@@ -544,6 +558,18 @@ def wrapper(*args):
     return wrapper
 
 
+def _auth_context_middleware(wsgi_app, auth_providers):
+    def wrapper(*args):
```

Comment:
Can we simply specify these arguments as `environ` and `start_response`?

I see that `_handling_errors` doesn't do that, I assume to avoid eating the `self` parameter, but I think that's only relevant when trying to make a middleware that can be used as a decorator on a WSGI-app-as-method (as with `wrappers.Request()`) which we haven't bothered doing for our standalone middlewares like PathPrefixMiddleware, so there doesn't seem to be a ton of reason to do it here.

Reply:
I think you’re right: `_handling_errors` was previously used as a
decorator, but we refactored that in #2732. Will change.

Reply:
Done, and likewise for `_handling_errors` in #3827.